markets/perps-market/contracts/storage/AsyncOrder.sol

```diff
@@ -369,6 +368,53 @@ library AsyncOrder {
         return (runtime.newPosition, runtime.orderFees, runtime.fillPrice, oldPosition);
     }
 
+    /**
+     * @notice Checks if the order request can be cancelled.
+     * @notice This function doesn't check for liquidation or available margin since the fees to be paid are small and we did that check at commitment less than the settlement window time.
+     * @dev it calculates fill price the order
+     * @dev and with that data it checks that:
+     * @dev - settlement window is open
+     * @dev - the fill price is outside the acceptable price range
+     * @dev if the order can be cancelled, it returns the fillPrice
+     */
+    function validateCancellation(
+        Data storage order,
+        SettlementStrategy.Data storage strategy,
+        uint256 orderPrice
+    ) internal view returns (uint256 fillPrice) {
+        SimulateDataRuntime memory runtime;
+        runtime.sizeDelta = order.request.sizeDelta;
+        runtime.accountId = order.request.accountId;
+        runtime.marketId = order.request.marketId;
+
+        // Validate Order still exists and is valid
+        if (runtime.sizeDelta == 0) {
+            revert AsyncOrder.ZeroSizeOrder();
+        }
+
+        checkWithinSettlementWindow(order, strategy);
```

Comment:
ok yeah this is what i mentioned above, as long as this is there, we should be fine